web/libs/core/src/lib/Tour/TourProvider.tsx

```diff
@@ -0,0 +1,181 @@
+ * to manage its state and behavior. The Tour component wraps react-joyride to render
+ * the actual tour UI elements.
+ */
+
+interface TourState {
+  key: string;
+  run: boolean;
+  continuous: boolean;
+  loading: boolean;
+  stepIndex: number;
+  steps: Step[];
+}
+
+export interface TourAction {
+  type: "START" | "RESET" | "STOP" | "GOTO" | "NEXT" | "RESTART";
+  payload?: any;
+}
+
+const reducer = (state: TourState, action: TourAction): TourState => {
+  switch (action.type) {
+    case "START":
+      return { ...state, run: true };
+    case "RESET":
+      return { ...state, stepIndex: 0 };
+    case "STOP":
+      return { ...state, run: false };
+    case "NEXT": {
+      const nextIndex = state.stepIndex + 1;
+      if (nextIndex < state.steps.length) return { ...state, stepIndex: nextIndex };
+      console.log(`Tour ${state.key} has no more steps`);
+      return state;
+    }
+    case "GOTO":
+      return { ...state, ...action.payload };
+    case "RESTART":
+      return {
+        ...state,
+        stepIndex: 0,
+        run: true,
+        loading: false,
+        key: new Date().toISOString(),
+      };
+    default:
+      return state;
+  }
+};
+
+export const userTourStateReducer = (): [TourState, React.Dispatch<TourAction>] =>
+  useReducer(reducer, createInitialState([]));
+
+const createInitialState = (steps: Step[]): TourState => ({
+  key: new Date().toISOString(),
+  run: false,
+  continuous: true,
+  loading: false,
+  stepIndex: 0,
+  steps: steps.map((step) => ({
+    ...step,
+    // TODO: although html is predefined by LSE assets, to avoid XSS, we should better sanitize it by allowing only simple html tags like <b>, <i>, <u> etc.
+    title: typeof step.title === "string" ? <div dangerouslySetInnerHTML={{ __html: step.title }} /> : step.title,
+    content:
+      typeof step.content === "string" ? <div dangerouslySetInnerHTML={{ __html: step.content }} /> : step.content,
+  })),
+});
+
+type ProductTourState = "ready" | "skipped" | "completed";
+
+const updateProductTourState = (
+  api: any,
+  name: string,
+  state: ProductTourState,
+  interactionData: Record<string, any> = {},
+) => {
+  return api.callApi("updateProductTour", {
+    params: {
+      name,
+    },
+    body: {
+      state,
+      interaction_data: interactionData,
+    },
+  });
+};
+
+interface TourContextType {
+  registerTour: (name: string, dispatch: React.Dispatch<TourAction>) => void;
+  unregisterTour: (name: string) => void;
+  startTour: (name: string) => void;
+  setTourViewed: (name: string, isSkipped: boolean, interactionData: Record<string, any>) => void;
+  restartTour: (name: string) => void;
+}
+
+export const TourContext = createContext<TourContextType | null>(null);
+
+export const TourProvider: React.FC<{
+  children: React.ReactNode;
+  useAPI: () => { callApi: (apiName: string, params: Record<string, any>) => any };
+}> = ({ children, useAPI }) => {
```

Comment:
passing useAPI is a good solution for now, but I think we should have a todo comment here to explain that we should clean this up once useAPI is properly unified and moved into core library
```suggestion
// TODO: once useAPI is unified and moved into core library we need to come back and clean this up as it will not be necessary to pass it in to the provider
export const TourProvider: React.FC<{
  children: React.ReactNode;
  useAPI: () => { callApi: (apiName: string, params: Record<string, any>) => any };
}> = ({ children, useAPI }) => {
```